### ISSUE TITLE
Remove pre-compilation of contracts/scripts for sncast tests

### ISSUE THREAD

**Issue by THenry14** [foundry-rs/starknet-foundry#1800]
In order to cache scarb artifacts we once created a `build_contracts.sh` script, that compiles contracts if there are no artifacts, and skips the compilation if artifacts are already present. The script is no longer needed, and in fact generates some problems when we update scarb. We should just get rid of it.

**Comment by THenry14** [comment]
blocked by migration to newer devnet version, where network will be forked.
